tensorboard/backend/process_graph.py

```diff
@@ -46,6 +46,9 @@ def prepare_graph_for_ui(
       ValueError: If `large_attrs_key is None` while `limit_attr_size != None`.
       ValueError: If `limit_attr_size` is defined, but <= 0.
     """
+    # TODO(@davidsoergel): detect whether a graph has been filtered already
```

Comment:
To make this TODO item clearer, I believe you can say in addition: "if it is already filtered, return immediately".

Reply:
Done.